src/web/components/CMP.tsx

```diff
@@ -1,35 +1,43 @@
 import React, { useState, useEffect } from 'react';
-import {
-    shouldShow,
-    setErrorHandler,
-} from '@guardian/consent-management-platform';
-import { ConsentManagementPlatform } from '@guardian/consent-management-platform/lib/ConsentManagementPlatform';
-
-type Props = {
-    cmpUi: boolean; // A switch to decide if we show CMP or not
-};
+import loadable from '@loadable/component';
+
+const ConsentManagementComponent = loadable.lib(() =>
+    import(
+        '@guardian/consent-management-platform/lib/ConsentManagementPlatform'
+    ),
+);
 
-export const CMP = ({ cmpUi }: Props) => {
```

Comment:
I feel like we shouls be using `cmpUI` here to block the import

Reply:
as in use `cmpUi` to dynamically import the component? This is what I originally did but did not see a substantial decrease in the bundle size. Thus why I ended up lazy loading the lib

Reply:
As far as I understand it, `cmpUI` could be renamed `cmpIsEnabled` (site wide) and then `shouldShow` is a local flag for that user/browser.

Based on that, we only need to import the lib if `cmpIsEnabled` and then within the lib/component there's local state to show hide based on `shouldShow`. But it's fair to assume it will always be enabled so the advantage to dynamically loading on this switch is, I think, that the loading is removed from the initial bundle and moved to App - but still immediate?

The next logical step from here would be to extract the `shouldShow` logic so that we can dynamically import based on if the user has seen the banner or not - that might need a refactor of CMP though

Reply:
I hope to get som clarification on what `cmpUi` means in this thread https://github.com/guardian/dotcom-rendering/pull/1351#discussion_r410088412. In the meantime I am just focused on not changing any of the condutional logic.

I fully agree, we should only  fetch the CMP and related libs if `cmpUi` is false. Thats is why I lifted the `cmpUi` condition logic to `App.tsx`, meaning none of the library is dynamically loaded if `cmpUi` is true. Also the component `ConsentManagementPlatform` only gets fetched if `show` is true. We therefore have dynamically loaded both the functions `shouldShow` and `setErrorHandler` (which are  ~50kb) and the component `ConsentManagementPlatform` (around ~20kb).

Reply:
We opted for not having the `shouldShow` within the component because the CMP UI can still be surfaced even if this function returns false (by the user clicking the privacy settings link at the bottom of the page). Having a separate shouldShow function also helps when determining if other banners can be shown or not.

Reply:
@ripecosta in other words `shouldShow` should always be fired irregardless of `cmpUi`?

Reply:
In this case, yes. You always need to check both to surface the UI.

Reply:
> I fully agree, we should only fetch the CMP and related libs if `cmpUi` is false. Thats is why I lifted the `cmpUi` condition logic to `App.tsx`, meaning none of the library is dynamically loaded if `cmpUi` is true.

I think the logic here is the opposite. We should only render `<CMP />` if cmpUI is `true`

Reply:
Correct. We should only render `<CMP />` when `cmpUI` is true and `shouldShow` returns true.

Reply:
thanks for catching this @ripecosta I now see i did in fact change the logic here. I will revert to a previous version.

Also I wanted to double cheack, accorinding to @GHaberis https://github.com/guardian/dotcom-rendering/pull/1351#discussion_r410088412 the `cmpUi` logic was wrongly implemented before, we want to render the `CMP` component if `cmpUi` is true?

Reply:
@ripecosta so just to confirm: `if(cmpUI && shouldShow() === true)`  then display `<CMP />` ? 

so its not an `||` condition but a `&&`?

Reply:
This was what we had initially: `<CMP cmpUi={CAPI.config.cmpUi} />` which was then being correctly checked inside the component:
```
if (!show || !cmpUi) {
    return null;
}
```

One of the versions on this PR changed the logic to `{CAPI.config.cmpUi === false && <CMP />}` which is incorrect, but has since been fixed to `{CAPI.config.cmpUi && <CMP />}`.

Extra points if you do something like `{CAPI.config.cmpUi && shouldShow() && <CMP />}` because in most pageviews `shouldShould()` will return false, despite the switch being enabled, because the UI won't need to be show.

Reply:
to clarify, negations make things hard to discuss ^^ here is some recap:
`{CAPI.config.cmpUi && <CMP />}` => we life the `CAPI.config.cmpUi` logic to `App.tsx`
`show` should be used to render the component, and not to return null (for clarity)